src/main/scala/mesosphere/marathon/core/deployment/impl/StartingBehavior.scala

```diff
@@ -46,7 +46,9 @@ trait StartingBehavior extends ReadinessBehavior with StrictLogging { this: Acto
     case InstanceChanged(id, `version`, `pathId`, condition: Condition, instance) if condition.isTerminal || instance.isReservedTerminal =>
       logger.warn(s"New instance [$id] failed during app ${runSpec.id.toString} scaling, queueing another instance")
       instanceTerminated(id)
-      launchQueue.add(runSpec, 1).pipeTo(self)
+      if (instance.state.goal != Goal.Running) {
+        launchQueue.add(runSpec, 1).pipeTo(self)
+      }
 
     case Sync => async {
       val instances = await(instanceTracker.specInstances(runSpec.id))
```

Comment:
Doesn't the same over-scaling problem apply here to the `Sync` message? Shouldn't we take into account instance goals when we calculate `instancesToStartNow`, @alenkacz ?

Reply:
here we work with `isScheduled` so we should be fine, that one already works with goals